### DISCUSSION TITLE
State unsafe mutation error with Tabs

### DISCUSSION THREAD

**Discussion by WarningImHack3r** [huntabyte/shadcn-svelte#1523] [Help] [answered]
I'm trying to upgrade [svelte-changelog](https://svelte-changelog.vercel.app) to Svelte 5, and I've got this error upgrading shadcn-svelte:

<img src="https://github.com/user-attachments/assets/e9d48ee3-4e90-40c9-ba68-9b0b3ed18367" alt="error" width="400" />

I'm not sure whether it comes from the `tabs-content.svelte` component or from my code, hence my post here.
The PR is here: https://github.com/WarningImHack3r/svelte-changelog/pull/35. Can anyone help me out, please?

**Comment by huntabyte** [comment]
What version of bits are you using?

**Reply by WarningImHack3r** [reply]
The latest beta one at the time (everything’s in the linked PR)
I've tried beta 28 first, then I'm now in beta 64, exact same issue

**Reply by huntabyte** [reply]
Can you destructure from a derived like this?? 
![CleanShot 2024-11-30 at 12 57 39@2x](https://github.com/user-attachments/assets/8b9520e4-52b0-4e61-ae4f-d3d8caa36f1d)

If so, TIL

I'm looking through the PR

**Reply by huntabyte** [reply]
That was an issue at one point but it was resolved. Try clearing your node modules to see if it persists.

**Reply by WarningImHack3r** [reply]
> Can you destructure from a derived like this?? If so, TIL

Yes, why couldn't you? :D it's very useful when you want to derive multiple things from a single "common derivation"

Edit: [grep.app](https://grep.app/search?q=%20%7D%20%3D%20%24derived) made me stumble upon [a test for that](https://github.com/sveltejs/svelte/blob/main/packages/svelte/tests/runtime-runes/samples/destructure-derived-object/main.svelte), and that feature is 100% supported since 5.0.0-next.69!

**Reply by WarningImHack3r** [reply]
> That was an issue at one point but it was resolved. Try clearing your node modules to see if it persists.

Alright I'm trying that

Edit: no luck, same problem

**Reply by WarningImHack3r** [reply]
(Up in case you didn’t see my edit; happy new year btw! No pressure, just adding a notification to your inbox so you can come back later)

**Comment by WarningImHack3r** [comment, marked as answer]
Update: it was my fault. As the error states, I was updating a state from the template (inside a `Tabs` content), which is (apparently) forbidden.
